src/sentry/workflow_engine/migration_helpers/issue_alert_migration.py

```diff
@@ -89,14 +99,23 @@ def _create_detector_lookup(self) -> Detector:
                 error_detector = Detector(type=ErrorGroupType.slug, project=self.project)
 
         else:
-            error_detector, _ = Detector.objects.get_or_create(
-                type=ErrorGroupType.slug,
-                project=self.project,
-                defaults={"config": {}, "name": "Error Detector"},
-            )
-            _, created = AlertRuleDetector.objects.get_or_create(
-                detector=error_detector, rule_id=self.rule.id
+            lock = locks.get(
+                f"workflow-engine-project-error-detector:{self.project.id}",
+                duration=10,
+                name="workflow_engine_issue_alert",
             )
+            with lock.acquire():
```

Comment:
are you thinking we'll want to have both a lock and the db constraint?

Reply:
i feel like i at least need the lock before putting in the DB constraint, otherwise we can get into a bad state again

Reply:
it does not hurt to have the lock as well